externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala

```diff
@@ -142,7 +151,18 @@ class ExecuteStatement(
           case TypedResult.ResultType.PAYLOAD =>
             (1 to result.getPayload).foreach { page =>
               if (rows.size < resultMaxRows) {
-                rows ++= executor.retrieveResultPage(resultId, page).asScala
+                val result = executor.retrieveResultPage(resultId, page).asScala.toList
+                result.headOption match {
+                  case None =>
+                  case Some(r) =>
+                    // for flink 1.14
+                    if (r.getClass == classOf[Row]) {
+                      rows ++= result.asInstanceOf[List[Row]]
+                    } else {
+                      // for flink 1.15+
+                      rows ++= result.map(r => convertToRow(r.asInstanceOf[RowData], dataTypes))
```

Comment:
From my understanding, `RowData` is more future proof than `Row`, then `convertToDataRow` should more reasonable?
@link3280 @SteNicholas

Reply:
IIUC, `RowData` is Flink's internal data type, thus I think `Row` is more suitable for our case.

Reply:
Are we suffering from [FLINK-21456](https://issues.apache.org/jira/browse/FLINK-21456)?

Reply:
I talk with Cheng Pan offline, LGTM.